app/services/log_event_searcher.rb

```diff
@@ -0,0 +1,60 @@
+class LogEventSearcher
+
+  ALLOWED_EVENTS = ["account.create", "account.update",
+                    "account_user.create", "account_user.delete",
+                    "user.create"].freeze
+
+  def self.beginning_of_time
+    10.years.ago
+  end
+
+  attr_accessor :start_date, :end_date, :events, :query
+
+  def initialize(start_date: nil, end_date: nil, events: [], query: nil)
+    @start_date = start_date
+    @end_date = end_date
+    @events = filter_events(events)
+    @query = query
+  end
+
+  def filter_events(events)
+    Array(events).select { |event| event.in?(ALLOWED_EVENTS) }
+  end
+
+  def search
+    result = LogEvent.all
+    result = result.merge(filter_date) if start_date || end_date
+    result = result.merge(filter_event) if events.present?
+    result = result.merge(filter_query) if query.present?
+    result
+  end
+
+  def dates
+    [(start_date || LogEventSearcher.beginning_of_time), (end_date || Date.current)]
+  end
+
+  def filter_date
+    LogEvent.where(event_time: (dates.min.beginning_of_day..dates.max.end_of_day))
+  end
+
+  # When this gets updated to Rails 5, you can use ActiveRecord#or directly
+  # the event name comes in as <loggable_type>.<event_type>
+  def filter_event
+    where_strings = events.flatten.map do |event|
+      loggable_type, event_type = event.split(".")
+      "(loggable_type = '#{loggable_type.camelize}' AND event_type = '#{event_type}')"
+    end
+    LogEvent.where(where_strings.join(" OR "))
+  end
+
+  def filter_query
+    account_ids = Account.where("account_number LIKE ?", "%#{query}%").pluck(:id)
```

Comment:
I'm thinking we should probably also include any `AccountUser`s that match the account query. That way if I search for chart string X, I get the creation/updates as well as the add/remove users for that account. Maybe we want to do the same for the users, but I care a little bit less about that. What do you think?